docs/content/developers/controllers/writing-kcp-aware-controllers.md

```diff
@@ -7,23 +7,33 @@ description: >
 
 # Writing kcp-aware Controllers
 
-## Keys for Objects in Listers/Indexers
+While kcp follows the Kubernetes Resource Model (KRM) closely, its multi-tenancy capabilities via workspaces / logical clusters
+make normal Kubernetes controllers somewhat limited in usefulness, since they generally expect a single Kubernetes API endpoint
+(which would correspond to a single logical cluster). While kcp exposes something called [virtual workspaces](../../concepts/workspaces/virtual-workspaces.md),
+their API semantics are also different from standard Kubernetes API endpoints and thus, controllers written for Kubernetes
+cannot be used against them.
 
-When you need to get an object from a kcp-aware lister or an indexer, you can't just pass the object's name to the
-`Get()` function, like you do with a typical controller targeting Kubernetes. Projects using kcp's copy of client-go
-are using a modified key function.
+!!! info
+    kcp has previously maintained a [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) fork under [kcp-dev/controller-runtime](https://github.com/kcp-dev/controller-runtime).
+    While it likely still works, we recommend migration to multicluster-runtime (see below).
 
-Here are what keys look like for an object `foo` for both cluster-scoped and namespace-scoped varieties:
+## multicluster-runtime
 
-|Organization|Workspace|Logical Cluster|Namespace|Key|
-|-|-|-|-|-|
-|-|-|root|-|root|foo|
-|-|-|root|default|default/root|foo|
-|root|my-org|root:my-org|-|root:my-org|foo|
-|root|my-org|root:my-org|default|default/root:my-org|foo|
-|my-org|my-workspace|my-org:my-workspace|-|my-org:my-workspace|foo|
-|my-org|my-workspace|my-org:my-workspace|default|default/my-org:my-workspace|foo|
+[multicluster-runtime](https://github.com/kubernetes-sigs/multicluster-runtime) is an effort initiated by kcp developers in the upstream
+community to provide a framework for building multi-cluster controllers. multicluster-runtime is **not** a fork of controller-runtime,
+instead it is a sort of "addon" built on top of controller-runtime's excellent Go generic support. While the project is relatively young
+and deemed experimental, we are confident in its ability to provide a great controller development experience for kcp.
 
-## Encoding/Decoding Keys
+The runtime itself is generic and needs a so-called provider that provides it with information about the fleet of clusters
+it is supposed to reconcile over. A small ecosystem of providers exists already. Check out the upstream repository for more information.
```

Comment:
I think this sentence could lead to confusion. I think readers of this doc are interested in kcp providers directly and the next paragraph is very interesting for them. I think we should not put them on the wrong trail by (potentially) indirecting them to go to multicluster provider first, scroll through the Readme, hoping they find the "Provider Ecosystem" section and then return back to this guide to finish our "Providers for kcp" section.

Instead I think it is better to leave this out, so readers get to the "Providers for kcp" section without being sidetracked.